diff --git a/dist/cli.d.ts b/dist/cli.d.ts
index c9a2c033db12ac528fc99269028c5227ccd8e971..6e6362e111da25bb33d382b0d83941001eb87666 100644
--- a/dist/cli.d.ts
+++ b/dist/cli.d.ts
@@ -1,5 +1,4 @@
 import { f as ViteNodeServerOptions } from './types-e8623e9c.js';
-import 'vite/types/hot';
 import './types.d-7442d07f.js';
 
 interface CliOptions {
diff --git a/dist/client.d.ts b/dist/client.d.ts
index 18dce84d926ec5688f60b8b12a31e4bd5844cfaf..0e7d39acc274f7ba2526c29431cf2d25172061ae 100644
--- a/dist/client.d.ts
+++ b/dist/client.d.ts
@@ -1,3 +1,2 @@
 import './types.d-7442d07f.js';
 export { i as DEFAULT_REQUEST_STUBS, M as ModuleCacheMap, h as ViteNodeRunner } from './types-e8623e9c.js';
-import 'vite/types/hot';
diff --git a/dist/hmr.d.ts b/dist/hmr.d.ts
index aa5259130deb4105e50b71d53f8d7b2e876c1000..7e9b94d3ed68bbe118a6b291729ec5672655a202 100644
--- a/dist/hmr.d.ts
+++ b/dist/hmr.d.ts
@@ -1,7 +1,6 @@
 import { EventEmitter } from 'node:events';
 import { HMRPayload as HMRPayload$1, Plugin } from 'vite';
 import { h as ViteNodeRunner, H as HotContext } from './types-e8623e9c.js';
-import 'vite/types/hot';
 import './types.d-7442d07f.js';
 
 type EventType = string | symbol;
diff --git a/dist/index.d.ts b/dist/index.d.ts
index cd87e80f206a965e4f8bcfa850a8549f3b61bc8c..94821f942da8915cf3ab20db776f2be02c62d80f 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,3 +1,2 @@
 export { A as Arrayable, a as Awaitable, C as CreateHotContextFunction, g as DebuggerOptions, D as DepsHandlingOptions, b as FetchFunction, F as FetchResult, H as HotContext, d as ModuleCache, M as ModuleCacheMap, N as Nullable, R as RawSourceMap, c as ResolveIdFunction, S as StartOfSourceMap, e as ViteNodeResolveId, V as ViteNodeRunnerOptions, f as ViteNodeServerOptions } from './types-e8623e9c.js';
 export { D as DecodedSourceMap, E as EncodedSourceMap } from './types.d-7442d07f.js';
-import 'vite/types/hot';
diff --git a/dist/server.d.ts b/dist/server.d.ts
index 84ec033f6959991d89c507ef6b874667fcdb5389..919ec623a0b0e80307fa2e7899cf0fb91957c902 100644
--- a/dist/server.d.ts
+++ b/dist/server.d.ts
@@ -1,7 +1,6 @@
 import { TransformResult, ViteDevServer } from 'vite';
 import { E as EncodedSourceMap } from './types.d-7442d07f.js';
 import { g as DebuggerOptions, D as DepsHandlingOptions, f as ViteNodeServerOptions, F as FetchResult, e as ViteNodeResolveId } from './types-e8623e9c.js';
-import 'vite/types/hot';
 
 declare class Debugger {
     options: DebuggerOptions;
diff --git a/dist/types-e8623e9c.d.ts b/dist/types-e8623e9c.d.ts
index dfe926f5997f14037211e1f8bf6aa84cd842d831..66e17d99bed1464dcf74b94c7830a40edb94cb03 100644
--- a/dist/types-e8623e9c.d.ts
+++ b/dist/types-e8623e9c.d.ts
@@ -1,4 +1,3 @@
-import { ViteHotContext } from 'vite/types/hot';
 import { E as EncodedSourceMap } from './types.d-7442d07f.js';
 
 declare const DEFAULT_REQUEST_STUBS: Record<string, Record<string, unknown>>;
@@ -107,7 +106,7 @@ interface FetchResult {
     externalize?: string;
     map?: EncodedSourceMap | null;
 }
-type HotContext = Omit<ViteHotContext, 'acceptDeps' | 'decline'>;
+type HotContext = any;
 type FetchFunction = (id: string) => Promise<FetchResult>;
 type ResolveIdFunction = (id: string, importer?: string) => Awaitable<ViteNodeResolveId | null | undefined | void>;
 type CreateHotContextFunction = (runner: ViteNodeRunner, url: string) => HotContext;
diff --git a/dist/types.d.ts b/dist/types.d.ts
index 600c5414edb6c88d55c31cfc6091da0fa1bea950..787c5323594158c03f43d8bb92d87b81057983b0 100644
--- a/dist/types.d.ts
+++ b/dist/types.d.ts
@@ -1,3 +1,2 @@
-import 'vite/types/hot';
 export { D as DecodedSourceMap, E as EncodedSourceMap } from './types.d-7442d07f.js';
 export { A as Arrayable, a as Awaitable, C as CreateHotContextFunction, g as DebuggerOptions, D as DepsHandlingOptions, b as FetchFunction, F as FetchResult, H as HotContext, d as ModuleCache, M as ModuleCacheMap, N as Nullable, R as RawSourceMap, c as ResolveIdFunction, S as StartOfSourceMap, e as ViteNodeResolveId, V as ViteNodeRunnerOptions, f as ViteNodeServerOptions } from './types-e8623e9c.js';
diff --git a/dist/utils.d.ts b/dist/utils.d.ts
index 4b9c0f4543811381a6e007678f6cefbd7d00913b..b491088eae2fac28765485487f94fe5b73b9de52 100644
--- a/dist/utils.d.ts
+++ b/dist/utils.d.ts
@@ -1,5 +1,4 @@
 import { N as Nullable, A as Arrayable } from './types-e8623e9c.js';
-import 'vite/types/hot';
 import './types.d-7442d07f.js';
 
 declare const isWindows: boolean;
diff --git a/package.json b/package.json
index 189926dad18c505ac6fc21baa1cdc8597b4c01c6..ab02d633b4bbac2d3ef623d22cbae2761ef96559 100644
--- a/package.json
+++ b/package.json
@@ -1,4 +1,5 @@
 {
+  "type": "module",
   "name": "vite-node",
   "version": "0.34.1",
   "description": "Vite as Node.js runtime",
